operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -20,15 +19,14 @@
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
 import io.javaoperatorsdk.operator.processing.event.EventSourceManager;
-import io.javaoperatorsdk.operator.processing.event.EventSourceRegistry;
+import io.javaoperatorsdk.operator.processing.event.source.EventSourceRegistry;
 
 public class Controller<R extends HasMetadata> implements Reconciler<R>,
     LifecycleAware, EventSourceInitializer<R> {
   private final Reconciler<R> reconciler;
   private final ControllerConfiguration<R> configuration;
   private final KubernetesClient kubernetesClient;
   private EventSourceManager<R> eventSourceManager;
-  private EventProcessor<R> eventProcessor;
```

Comment:
The idea as desribed here is that the controller maneges the lifecycle of every other component top level component:
https://github.com/java-operator-sdk/java-operator-sdk/issues/655

So it is the highest level agregate, therefore will be no multiple layers for managing LifecycleAware

Reply:
The problem is that it doesn't really do anything with it, it's just needed by the event manager so I don't think we should expose it since the lifecycle of the `EventProcessor` is tied to the `EventManager` (you cannot have one without the other). Making `EventManager` handle the `EventProcessor` life cycle also allows to avoid the rather ugly setters and prevent forgetting to start/stop one or the other. 
If the `Controller` was using the `EventProcessor` directly, it would make sense to keep there. At the very least, if we wanted to keep the top-level aggregate concept, the `EventManager` should retrieve the `EventProcessor` from the `Controller` (or vice-versa) and that's actually what I tried doing yesterday without complicating things even more…

Reply:
But in general the Controller does not do anything just instantiates and starts/stops the major components. The circular dependency is ugly, but the EventManager is just a class to encapsulate a list of EventSources. For me that it creates a central component, probably the most complex part is quite a smell.

Reply:
Except that `EventSourceManager` and `EventProcessor` are not really separate components. Again, one cannot exist without the other. If we want to keep them separate, we should make it as cleanly as possible and I think the current solution is cleaner because it avoids the mess of having incomplete objects for no good reason. Maybe we should combine `EventSourceManager` and `EventProcessor`, renaming them to `EventManager`?

Reply:
The problem with that is even now the EventProcessor is too big, with bunch of logic, what little bugs me. Actually wanted to somehow break it down to more components.

Reply:
It is indeed quite complex, but some things are 😄
It doesn't make sense to split things that are not coherent if we end up with more complexity to manage the different pieces. I could see the `TimerEventSource` moving back to `EventSourceManager` but I really think that we should tie the lifecycle of `EventSourceManager` and `EventProcessor` by making `EventSourceManager` handle the processor's lifecycle because, again, the manager cannot work without the processor.

Reply:
Then the event source manager will handle the lifecycle of event processor? 
ok that is fine by me, we can evaluate further later.